server/util/tokenBalanceCache.ts

```diff
@@ -75,48 +76,47 @@ export default class TokenBalanceCache extends JobRunner<CacheT> {
       }));
 
     try {
-      const tokensFromListsResponses = await models.Token.findAll();
-      const tokensFromLists: TokenForumMeta[] = tokensFromListsResponses
-        .map((o) => {
-          return {
-            id: slugify(o.name),
-            address: o.address,
-            name: o.name,
-            symbol: o.symbol,
-            iconUrl: o.logoURI,
-          };
-        });
-
-      return [...tokens, ...tokensFromLists];
+      return tokens.concat(
+        (await this.models.Token.findAll()
+          .map((o) => {
+            return {
+              id: slugify(o.name),
+              address: o.address,
+              name: o.name,
+              symbol: o.symbol,
+              iconUrl: o.logoURI,
+            };
+          }))
+      );
     } catch (e) {
       log.error('An error occurred trying to access token lists', e.message);
     }
 
     return tokens;
   }
 
-  public getToken(searchAddress: string): TokenForumMeta {
-    return this._contracts.find(({ address }) => address === searchAddress);
+  public getToken(searchAddress: string): Promise<TokenForumMeta> {
+    return this.getTokens().then((o) => o.find(({ address }) => { return address.toLowerCase() === searchAddress.toLowerCase(); }));
   }
 
   public async start(models?, prefetchedTokenMeta?: TokenForumMeta[]) {
+    let tokenMeta;
     if (!prefetchedTokenMeta) {
-      const tokenMeta = await this._connectTokens(models);
-      this._contracts = tokenMeta;
+      tokenMeta = await this.getTokens();
```

Comment:
Similarly, since we have all the tokens in the database already here, we should only create an object on tokenMeta when a request is actually made, rather than instantiating all tokens to empty objects. Goal being to avoid huge memory usage of maintaining 30k empty objects.